lib/resources/xinetd.rb

```diff
@@ -75,8 +75,13 @@ def read_params
           params[k] = v
         else
           params['services'][name] = v
+
           # add the service identifier to its parameters
-          v.each { |service| service.params['service'] = name }
```

Comment:
Minro style point.  You might consider avoidng the type check and doing:

```
Array(v).each { |service| service.params['service'] = name }
```

Array(v) will coerce a scalar into an array but won't touch the arg if it is already an array:

```
Airb(main):001:0> Array(4)
=> [4]
irb(main):002:0> Array([4])
=> [4]
```